From 5bd0d02a8402a793adf48d06ac4bead1edca9a02 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 18 Nov 2019 16:29:48 +0200
Subject: [PATCH] [SW-6739] hl: add return point after pcie init

In case we are not about to proceed to preboot just return to caller

This is useful for goya2 when arcboot_rom uses ppboot to establish pcie
link

Change-Id: I21c39f3b04b4afe279787bf7d1f6aed791ccf361
---

diff --git a/app/ppboot.c b/app/ppboot.c
index d03695a..6f6c4cb 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -93,6 +93,11 @@
 	preboot_sync_btl();
 	hl_uart_init();
 	ppboot_pcie_init();
+
+	/* check if we are done here */
+	if (hl_pcie_init_only())
+		return;
+
 	preboot_delay();
 	copy_preboot();
 
diff --git a/gaudi/common.c b/gaudi/common.c
index d5bb4af..f3652b5 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -186,3 +186,8 @@
 	while (regmap_readl(mmPSOC_GLOBAL_CONF_BTM_FSM))
 		;
 }
+
+int hl_pcie_init_only(void)
+{
+	return 0;
+}
diff --git a/goya/common.c b/goya/common.c
index 4005bd2..94e6723 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -169,3 +169,8 @@
 
 void hl_pll_wa(void) {}
 void preboot_sync_btl(void) {}
+
+int hl_pcie_init_only(void)
+{
+	return 0;
+}
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 63f9229..3549cc6 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -139,3 +139,9 @@
 
 void hl_pll_wa(void) {}
 void preboot_sync_btl(void) {}
+
+
+int hl_pcie_init_only(void)
+{
+	return 1;
+}
diff --git a/include/preboot.h b/include/preboot.h
index 22f57a9..1231e2f 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -80,6 +80,7 @@
 u32 hl_skip_pcie_init(void);
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
+int hl_pcie_init_only(void);
 
 void preboot_pcie_wait_for_link(void);
 void preboot_delay(void);
